src/DeploymentUtils.ts

```diff
@@ -1,19 +1,47 @@
-import * as deployments from "../deployments/deployments.json";
+import * as deployments_ from "../deployments/deployments.json";
+interface DeploymentExport {
+  [chainId: string]: { [contractName: string]: { address: string; blockNumber: number } };
+}
+const deployments: DeploymentExport = deployments_ as any;
 
 // Returns the deployed address of any contract on any network.
 export function getDeployedAddress(contractName: string, networkId: number): string {
   try {
-    return (deployments as any)[networkId.toString()][contractName].address;
+    return deployments[networkId.toString()][contractName].address; // First try match directly on the name provided.
   } catch (_) {
+    try {
+      // If this does not match then try search on a Regex on SpokePool to let the caller exclude the chain name.
+      for (const _contractName of Object.keys(deployments[networkId.toString()]))
```

Comment:
I think this is confusing. Can we just make aliases in the generation? Aka save an identical deployment under a different name using deployments.save?

Something like:

```ts
  const spokePoolDeployment = await deploy("Optimism_SpokePool", {
    from: deployer,
    log: true,
    skipIfAlreadyDeployed: true,
    args: [
      hubPool.address, // Set hub pool as cross domain admin since it delegatecalls the Optimism_Adapter logic.
      hubPool.address,
      "0x0000000000000000000000000000000000000000", // timer
    ],
  });
  await hre.deployments.save("SpokePool", spokePoolDeployment);
```

Thoughts?

Reply:
I don't think this will break anything. We're just adding a _second record_ of the same contract.

So you'll have "Polygon_SpokePool" and "SpokePool" on polygon with all the same details.

Hardhat will still have the artifact it expects, but we will also export a second artifact under a more generic name.